doc: wifi: Improve debugging (#21404)
nrfconnect/sdk-nrf: doc/nrf/protocols/wifi/debugging.rst

WPA supplicant debug logs
=========================

The WPA supplicant manages the Wi-Fi® connection and is responsible for the 802.11 protocol.
To debug issues related to the Wi-Fi connection, such as failure to initiate, failed connections, or disconnections, you can enable WPA supplicant debug logs.
To build with WPA supplicant debug logs enabled, run the following commands:

With west

 .. code-block:: console

    west build -p -b nrf7002dk/nrf5340/cpuapp samples/wifi/shell -- -Dnrf_wifi_shell_SNIPPET="wpa-supplicant-debug"

With CMake

 .. code-block:: console

    cmake -GNinja -Bbuild -DBOARD=nrf7002dk/nrf5340/cpuapp -Dnrf_wifi_shell_SNIPPET="wpa-supplicant-debug" samples/wifi/shell
    ninja -C build

.. note::

   By default, this feature enables the :kconfig:option:`CONFIG_LOG_MODE_IMMEDIATE` Kconfig option, which can help prevent log buffer overflows.
   However, it may impact system timing and performance.
   If you experience issues with the system timing or performance, you can disable this option by setting the :kconfig:option:`CONFIG_LOG_MODE_IMMEDIATE` Kconfig option to ``n`` in your project configuration file.

Driver debug logs
=================

> The nRF Wi-Fi driver is manages the data path and control path between the host and the nRF70 firmware.

Reviewer: ```suggestion
The nRF Wi-Fi driver manages the data path and control path between the host and the nRF70 firmware.
```

> For debugging any issues related to the data path and control path, for example, not receiving or sending packets, failures in sending commands, or not receiving events, you can enable driver debug logs.

Reviewer: ```suggestion
To debug issues related to the data path and control path, such as not receiving or sending packets, failures in sending commands, or not receiving events, you can enable driver debug logs.
```
